ArchiSteamFarm/Json/Steam.cs

```diff
@@ -476,16 +480,171 @@ internal sealed class InventoryResponse : NumberResponse {
 			private InventoryResponse() { }
 
 			internal sealed class Description {
+				[JsonExtensionData]
+				internal readonly ImmutableDictionary<string, JToken> AdditionalProperties;
+
 				[JsonProperty(PropertyName = "appid", Required = Required.Always)]
 				internal readonly uint AppID;
 
 				internal ulong ClassID { get; private set; }
 				internal ulong InstanceID { get; private set; }
 				internal bool Marketable { get; private set; }
-				internal Asset.ERarity Rarity { get; private set; }
-				internal uint RealAppID { get; private set; }
+
+				internal Asset.ERarity Rarity {
+					get {
+						Asset.ERarity rarity = Asset.ERarity.Unknown;
+
+						foreach (Tag tag in Tags) {
+							switch (tag.Identifier) {
+								case "droprate":
+									switch (tag.Value) {
+										case "droprate_0":
+											rarity = Asset.ERarity.Common;
```

Comment:
This is nice, it looks now how I've imagined it initially.

Now you can slightly improve performance of those getters, you can right away return the appropriate value instead of saving it, and falling back to return Unknown or other default in case no tag matches the expected name.

Of course only where it makes sense, `Asset.EType Type` can't be done this way since it needs to evaluate two, but `RealAppID ` and `Rarity` can.